### ISSUE TITLE
Move project website to eclipse.org/vertx

### ISSUE THREAD

**Issue by waynebeaton** [vert-x3/eclipse-proposal-wip#6]
The project's website should be moved to http://www.eclipse.org/vertx

All existing web properties (e.g. vertx.io) should redirect here.

Note that the custom website is still possible. In the absence of a website implementation, the current URL redirects to the automatically-generated "PMI" website.

**Comment by cescoffier** [comment]
What are the hosting features ? Can we run a docker container ? A JVM ? What about file access / upload ?

Even if our web site is very simple right now, we have on the roadmap improvements requiring more advanced features. 

Something we can do without any issue is to redirect http://www.eclipse.org/vertx to a server we manage (as http://vertx.io). 

It would make sense to me to keep the PMI web site for more formal information about the project. Would this still be possible ?

**Comment by pmlopes** [comment]
I can see that the project websites at eclipse all follow the same template, this seems nice for eclipse since all projects are organized the same way so a visitor of eclipse website will feel its navigation natural.

If I've to look at current vert.x website we provide a different kind of information, sure we provide releases and docs but we also include a blog, javadocs and as @cescoffier said we were considering of having live examples which would require a real JVM running, etc...

A quick look at the site also shows that this is probably not a hard requirement, e.g.: Eclipse IoT, a look at http://eclipse.org/iot returns a 404 the project description has a link to a `Website` which redirects to a different host under the DNS name `iot.eclipse.org`. I think we could have the same, a landing page such: http://projects.eclipse.org/projects/iot and redirect to http://vertx.io

**Comment by purplefox** [comment]
Oh no, please do not do this. It would be a hugely regressive move.

Vertx.io is a key part of the Vert.x brand. We have an extremely cool website which is streets ahead of the Eclipse template driven sites.

Converting to a "standard" Eclipse site would be a huge step backwards!

**Comment by purplefox** [comment]
I agree with clement - I think redirecting http://www.eclipse.org/vertx to a server we manage (as http://vertx.io) makes a lot more sense. That means those few people who do somehow end up on the Eclipse Vert.x page don't scratch their heads and wonder where the real web-site is.

**Comment by vietj** [comment]
Some projects on Eclipse.org have their own branding (https://eclipse.org/Xtext/), so we are aware we have some customization capabilities. However I don't think that being an Eclipse project should prevent us to have our own vertx.io website that we manage.

If the current eclipse.org/projects/vertx is unpleasant, we should for sure improve it and perhaps provide an adapted lightweight copy of vertx.io with more organizational info about the project, or a redirection to vertx.io .

I think we can be Eclipse Vert.x and retain control on our website because this is an important part of being a competitive open source project in 2016. Otherwise you remove from Vert.x an important asset and hinder its competitiveness.

**Comment by purplefox** [comment]
I don't think it makes any sense to maintain two web-sites. We have a great web-site which a lot of work went into and which does that job well.

If it's not broken, why fix it?

**Comment by maxandersen** [comment]
@purplefox there is one thing that is broken and that is eclipse.org/vertx - its a wasteland, I hope we are all okey fixing that :)

I don't think there are anyone saying vertx wont have control over the content and what is served on it. i.e. eclipse orion has a running vm for its demo site, i.e. https://orionhub.org/ also eclipse.org/xtext has own branding etc. with some basic requirements on ensuring the eclipse project management links are accessible from it etc.

I think sites like eclipse.org/xtext that "pops out" and has its own style/identity is important. But is it absolutely important wether the final page base url is vertx.io ?  (personally I don't give minus points to a project just because it is a sub page of something - it is the content and "uptodateness" of the page that counts to me) 

Making eclipse.org/vertx at least have content I think is good - but there are a few ways to do that.

a) eclipse.org/vertx redirect to vertx.io
b) vertx.io redirect to eclipse.org/vertx (and still use the same content and mechaism to generate it as before)
c) having vertx.io and eclipse.org/vertx contain the same content but just generated with two different base urls.

if you can still host the exact same things and there is a way to setup things like orionhub.org would that not be ok ? i.e. the only thing changing is that the base url is eclipse.org/vertx as opposed to vertx.io ?

**Comment by purplefox** [comment]
I don't see any need to change vertx.io, as I said before - if it's not broken don't fix it!

I think the only change should be in eclipse.org/vertx which should redirect to vertx.io.

**Comment by waynebeaton** [comment]
My apologies for the delay.

The primary project website needs to be at eclipse.org/vertx. This needs to be the go-to website for information about contributing, content for committers and adopters, and downloads. The reason is simple: independence. The content on the Eclipse servers is owned by the Eclipse Foundation, not an individual or company. I'm not even sure who owns the keys for vertx.io.

vertx.io can redirect to the eclipse.org/vertx or, you can set it up as a user portal as an augment to the content on eclipse.org/vertx

FWIW, I'm pretty sure that the content on vertx.io as it sits today can just be rehosted on the eclipse server (you don't have to conform to any specific layout or technology). The webpage footers may require some adjustment to include e.g. EF terms of use and copyright notices.

**Comment by purplefox** [comment]
> >  I'm not even sure who owns the keys for vertx.io.

You do. And it's somewhat concerning that you're not aware of the fact!

**Comment by purplefox** [comment]
Denis Roy is in control of the DNS for vertx.io - you can contact him for further details.

**Comment by waynebeaton** [comment]
> You do. And it's somewhat concerning that you're not aware of the fact!

Calm your fears. The Eclipse Webmaster owns the name and knows where it points.

Where is the website content source?

**Comment by vietj** [comment]
The website is a GitHub repository : https://github.com/vert-x3/vert-x3.github.io

It provides a simple, easy and powerful way to update the content.

**Comment by vietj** [comment]
The Eclipse Golo project is allowed to have its own website and not a redirection to the eclipse.org website: http://golo-lang.org

This website is hosted by a GitHub repository like Vert.x : https://github.com/golo-lang/golo-lang.github.com

Exactly the same situation than for Vert.x

**Comment by waynebeaton** [comment]
Thanks for pointing that out. I'll connect with the Golo project to ensure that they also conform.

**Comment by jponge** [comment]
@waynebeaton I think you have some email from me regarding Golo and its possible graduation that hasn't been answered. We're doing our best to play by the Eclipse rules and cope with issues like slow CQ processing, but sometimes I feel like we are not _as important_ as some projects can be. We are still looking for concrete support actions from the Foundation. Efforts should be done by all parties I think.

Anyway we had it granted that `golo-lang.org` was transferred to the Eclipse Foundation ownership, that the Foundation would deal with the DNS and that we could keep on publishing the pages with GitHub pages on `golo-lang.org`.

We can certainly get to compliance with the footers to put it in like with the Foundation requirements, like linking to the Copyright Agent and so on.

But let's discuss it on `golo-dev` and not pollute our friends at vert.x.

**Comment by purplefox** [comment]
"Make sure they conform". That statement would be hilarious if it wasn't so desperately sad. This isn't Soviet Russia.

It pretty much sums out how out of touch Eclipse is with how to foster dynamic communities in the 21st century.

I'm just glad I don't have to deal with this on a day to day basis any more. But I feel sorry for Julien and the rest of the team who are getting beaten with the stick now.

**Comment by maxandersen** [comment]
@jponge please feel free to ping me about issues around Golo and I'll help where I can.

**Comment by jponge** [comment]
@maxandersen :+1:
